api/src/cctp_calls.rs

```diff
@@ -78,6 +80,32 @@ pub fn schnorr_verify_signature(
     SchnorrSigScheme::verify(&FieldBasedSchnorrPk(pk.into_projective()), *msg, signature)
 }
 
+pub fn schnorr_derive_key_from_seed(seed: &[u8]) -> (SchnorrPk, SchnorrSk) {
+    // zero just default to random,
+    // however, is there a minimum length that should be required?
+    if seed.len() == 0 {
+        return schnorr_generate_key();
+    }
+
+    // Domain separation tag
+    const DST: &[u8] = &[0xFFu8; 32];
+
+    // Hash first to ensure size an eliminate any bias
+    // that may exist in `seed`
+    let mut hasher = blake2::Blake2b::default();
+    hasher.input(DST);
+    hasher.input(seed);
+    let digest = hasher.fixed_result();
+    let rng_seed = <[u8; 32]>::try_from(&digest[..32]).unwrap();
+    let mut rng = ChaChaRng::from_seed(rng_seed);
```

Comment:
Is ChaChaRng secure enough ? Quoting from crates.io:

> The ChaCha algorithm is widely accepted as suitable for cryptographic purposes, **but this implementation has not been verified as such**. Prefer a generator like OsRng that defers to the operating system for cases that need high security.

Reply:
Its used by the Linux Kernel which is periodically seeded from the kernel entropy pool. The only difference here is the seed vs using OsRng which uses ChaCha from /dev/random. We can yell at the caller with comments about how the seed should be long enough like 32 bytes.